doc/rst/technotes/libraries.rst

```diff
@@ -551,6 +558,149 @@ include the chapel runtime and standard modules for use in a non-Chapel program
+
+``CHPL_ZMQ_HOME`` would be set to ``/absolute/path/to/.local/``.
+
+Initializing Your Multilocale Library
+-------------------------------------
+
+Multilocale libraries can be used in a manner similar to single locale
+libraries.  However, as with transitioning between a single locale executable
+and a multilocale one, it is necessary to specify the number of locales required
+for the multilocale library.
+
+In C
+~~~~
+
+Users must still call ``chpl_library_init()`` before utilizing the exported
+Chapel functions.  However, the ``char* argv[]`` must now include two additional
+entries: the numlocales flag and its intended value.
+
+This can be accomplished either by explicitly adding the arguments in the C
+client program itself, or by passing them as arguments to the executable.
+
+This example demonstrates explicitly adding the arguments in the program using
+the ``foo`` library.
+
+.. code-block:: C
+
+   #include "foo.h"
+
+   int main(int argc, char* argv[]) {
+     int argChapelC = 3;
+     char* argChapelV[3] = {argv[0], "-nl", "2"};
+     // Initialize the Chapel runtime and standard modules
+     chpl_library_init(argChapelC, argChapelV);
+
+     baz(7); // Call into a library function
+
+     chpl_library_finalize();
+
+     return 0;
+   }
+
+Alternatively, the original single locale client from `Initializing Your
+Library In C`_ can be used with the additional two arguments to the executable:
+
+.. code-block:: bash
+
+   ./a.out -nl 2
+
+Users also still need to call the generated module initialization function for
+multilocale libraries, as mentioned in that section.
+
+In Python
+~~~~~~~~~
+
+Users must still call ``chpl_setup()`` before utilizing the exported Chapel
+functions.  However, it requires a ``numLocales`` argument when the library
+has been compiled for multilocale settings.  E.g. to run with ``4`` locales,
+write ``chpl_setup(4)``.
+
+Makefile-less Compilation
+-------------------------
+
+When compiling a C program using a multilocale Chapel library without a
+makefile, some additional steps are needed beyond those required for using a
+:ref:`single locale Chapel library <Makefileless Compilation In Single Locale>`.
+For this reason, we strongly suggest using the Makefile generated by the
+``--library-makefile`` flag, as described :ref:`here <Makefile Helper>`.
+
+A new ``compileline`` tool, ``compileline --multilocale-lib-deps`` is currently
+required after ``compileline --libraries``.  Additionally, the library being
+linked must be listed twice in the compilation command - once before
+``compileline --libraries`` and once at the end of the command.  The compilation
+command would then look like this (replacing ``myCProg.c`` with the name of your
+C program that will use the library):
+
+.. code-block:: sh
+
+   `$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries` `$CHPL_HOME/util/config/compileline --multilocale-lib-deps` -lfoo
```

Comment:
I find the `$()` and `${}` syntax clearer, but I'll leave that up to you. Also, newlines may be helpful here:

```suggestion
   $(${CHPL_HOME}/util/config/compileline --compile) myCProg.c \
   -Llib/ -lfoo $(${CHPL_HOME}/util/config/compileline --libraries) \
   $(${CHPL_HOME}/util/config/compileline --multilocale-lib-deps) -lfoo
```